modules/core/src/test/scala/io/janstenpickle/hotswapref/HotswapRefSpec.scala

```diff
@@ -48,11 +48,11 @@ class HotswapRefSpec extends AnyFlatSpec with Matchers with TestInstances {
       )
       _ <- Resource.eval(IO.sleep(1.second)) // give it a chance to update
       _ <- Resource.eval(hotswap.swap(Resource.pure(ref1)).start)
-      _ <- Resource.eval(IO.sleep(1.second)) // give it a chance to update
+      _ <- Resource.eval(IO.sleep(1.second)) // give it a chance to allocate a new resource and refresh the holder ref
       _ <- Resource.eval(hotswap.access.use(_.update("test1" :: _)))
     } yield (ref0.get, ref1.get, releaseSignal.get)).use(_.tupled)
 
-    val res = test.unsafeRunSync()
+    val res = (test <* IO.sleep(10.seconds)).unsafeRunSync()
```

Comment:
I can't seem to wrap my head around this sleep here. Is there a specific purpose for this?

Reply:
This is a leftover from a temporary version of this test, where I tracked the release of the resource. 

Yes, I agree that these tests should be reworked.